### PR TITLE
Yayson definitions

### PR DESCRIPTION
**Pull request by Codesleuth**
Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

Travis build: https://travis-ci.org/Codesleuth/DefinitelyTyped/builds/162540338

### CODE REVIEW

**Comment by Codesleuth** [comment]
It seems the bot stalled on the Travis-CI test stage.

File: yayson/yayson.d.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
+// Type definitions for yayson v2.0.1
+// Project: https://github.com/confetti/yayson
+// Definitions by: David Wood <https://github.com/Codesleuth>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare module 'yayson' {
+  class Store {
+    sync(obj: Object): void;
```
Comment (andy-ms) [inline yayson/yayson.d.ts]: This type definition doesn't make sense. If it returns `void` when you pass in an object, how can it also return a `T` for the exact same parameter list? Prefer to use `sync(obj: {}): any` or `sync(obj: {}): {}`, and downcast the result.

File: yayson/yayson.d.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    sync<T>(obj: Object): T;
+
+    find(type: string, id: string): any;
+    find<T>(type: string, id: string): T;
+
+    findAll(type: string): any[];
+    findAll<T>(type: string): T[];
+
+    remove(type: string, id: string): void;
```
Comment (andy-ms) [inline yayson/yayson.d.ts]: These can combine into `remove(type: string, id?: string): void`.

File: yayson/yayson.d.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    remove(type: string): void;
+
+    reset(): void;
+
+    records: Array<any>;
```
Comment (andy-ms) [inline yayson/yayson.d.ts]: Prefer `any[]`.

File: yayson/yayson.d.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    id(model: any): string;
+  }
+
+  class Presenter {
+    static adapter: string;
+
+    static render<T>(instanceOrCollection: PromiseLike<T>, options: y.JsonOptions): PromiseLike<T>;
+    static render<T>(instanceOrCollection: PromiseLike<T>): PromiseLike<T>;
+
+    static render(instanceOrCollection: any, options: y.JsonOptions): any;
```
Comment (andy-ms) [inline yayson/yayson.d.ts]: Use `options?: y.JsonOptions`

File: yayson/yayson.d.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    remove(type: string): void;
+
+    reset(): void;
+
+    records: Array<any>;
+    relations: { [key: string]: any };
+  }
+
+  interface Adapter {
+    new (): any;
```
Comment (andy-ms) [inline yayson/yayson.d.ts]: As written this would make individual instances of `Adapter` new-able. Did you mean `class Adapter` with `constructor();`?

Comment (Codesleuth) [inline yayson/yayson.d.ts]: This isn't needed, I'll remove it. It _is_ newable but it's never intended to be used that way.

File: yayson/yayson.d.ts (hunk 1 above)
Comment (Codesleuth) [inline yayson/yayson.d.ts]: Yep - mistake. It should be returning the thing that was sync'd.
Why would you use `{}` in the parameter list and not `any`?

File: yayson/yayson.d.ts (hunk 2 above)
Comment (Codesleuth) [inline yayson/yayson.d.ts]: Yeh I'll combine them all.

File: yayson/yayson.d.ts (hunk 1 above)
Comment (andy-ms) [inline yayson/yayson.d.ts]: I think they're identical in this context. (The difference would be that `any` isn't typechecked in the function's body, but this has no body.) Don't know which one is preferred.

**Review by ghost — APPROVED** [review]

**Pull request merged**